rust_snuba/src/strategies/clickhouse/batch.rs

```diff
@@ -79,17 +80,33 @@ impl BatchFactory {
             let client = self.client.clone();
 
             let result_handle = self.handle.spawn(async move {
-                let res = client
-                    .post(&url)
-                    .query(&[("query", &query)])
-                    .body(reqwest::Body::wrap_stream(UnboundedReceiverStream::new(
-                        receiver,
-                    )))
-                    .send()
-                    .await?;
-
-                if res.status() != reqwest::StatusCode::OK {
-                    anyhow::bail!("error writing to clickhouse: {}", res.text().await?);
+                while receiver.is_empty() && !receiver.is_closed() {
+                    // continously check on the receiver stream, only when it's
+                    // not empty do we write to clickhouse
+                    sleep(Duration::from_millis(800)).await;
+                }
+
+                if receiver.is_closed() && receiver.is_empty() {
+                    // batch is finished and we never had any data to write
+                    // to clickhouse
+                    return Ok(());
+                }
+
+                if !receiver.is_empty() {
```

Comment:
is this branch necessary, can't it run unconditionally?

Reply:
leaving this as is since if it's closed and empty we don't need to send a request to clickhosue